extensions/applicationinsights-osplugin-js/src/OsPlugin.ts

```diff
@@ -0,0 +1,253 @@
+                                    _os = _platformVersionResponse.platform;
+                                    _osVer = parseInt(_platformVersionResponse.platformVersion);
+                                    if (_os === "Windows"){
+                                        if (_osVer == 0){
+                                            _osVer = 8;
+                                        } else if (_osVer < 13){
+                                            _osVer = 10;
+                                        } else{
+                                            _osVer = 11;
+                                        }
+                                    }
+                                    sessionStorage.setItem("ai_os", _os);
+                                    sessionStorage.setItem("ai_osVer", _osVer.toString());
+                                }
+                            } else {
+                                _throwInternal(_core.logger,
+                                    eLoggingSeverity.CRITICAL,
+                                    _eInternalMessageId.PluginException,
+                                    "Could not retrieve operating system: " + response.reason);
+                            }
+                            _completeOsRetrieve();
+                        });
+                    }
+                }
+            }
+        
+            /**
+            * Complete retrieving operating system info process, cleanup and release the event queue
+            */
+            function _completeOsRetrieve() {
+                if (_getOSTimeout) {
+                    clearTimeout(_getOSTimeout);
+                }
+                _getOSInProgress = false;
+                _releaseEventQueue();
+            }
+        
+            /**
+            * Release internal event queue
+            */
+            function _releaseEventQueue() {
+                arrForEach(_eventQueue, (evt) => {
+                    if (_retrieveFullVersion){
+                        if (_endpointIsBreeze){
+                            setValue(getSetValue(evt.item, strExt), Extensions.OSExt, _os + _osVer);
+                        } else {
+                            setValue(getSetValue(evt.item, strExt), Extensions.OSExt, _os);
+                            setValue(getSetValue(evt.item, strExt), "osVer", _osVer);
+                        }
+                    }
+                    evt.ctx.processNext(evt.item);
+                });
+                _eventQueue = [];
+            }
+
+            function _initDefaults() {
+                _core = null;
+                _ocConfig = null;
+                _getOSInProgress = false;
+                _getOSTimeout = null;
+                _getOSTimeoutMs = null;
+                _retrieveFullVersion = false;
+                _eventQueue = [];
+                _retryTime = 0;
+            }
+
+            // For backward compatibility
```

Comment:
There is no backward compatibility required here.

If you need them for testing then use the debug hook rather than exposing the internal values